### DISCUSSION TITLE
How to use base64encode insise element

### DISCUSSION THREAD

**Discussion by amasahara** [renggli/dart-xml#118] [Q&A] [answered]
I try make an XML like this. But i don't know how to use with bulder.element
```
<?xml version="1.0"?>
    <bookshelf>
      <book>
        PHRpdGxlIGxhbmc9ImVuZ2xpc2giPkdyb3dpbmcgYSBMYW5ndWFnZTwvdGl0bGU+CiAgPHByaWNlPjI5Ljk5PC9wcmljZT4=
      </book>
      <book>
        PHRpdGxlIGxhbmc9ImVuZ2xpc2giPkxlYXJuaW5nIFhNTDwvdGl0bGU+CiAgICAgICAgPHByaWNlPjM5Ljk1PC9wcmljZT4=
      </book>
      <price>132.00</price>
    </bookshelf>
```

and data is

```
<?xml version="1.0"?>
    <bookshelf>
      <book>
        <title lang="english">Growing a Language</title>
        <price>29.99</price>
      </book>
      <book>
        <title lang="english">Learning XML</title>
        <price>39.95</price>
      </book>
      <price>132.00</price>
    </bookshelf>
```

**Comment by renggli** [comment, marked as answer]
There is a [tutorial](https://github.com/renggli/dart-xml#building) and [countless examples in the tests](https://github.com/renggli/dart-xml/blob/main/test/builder_test.dart) of how to use the builder.

To create a book element with arbitrary string data use: `builder.element('book', nest: base64encoded)`, where `base64encoded` is the String of your encoded data.

**Comment by amasahara** [comment]
thanks for that i has been do it already
